src/HeaderCell.tsx

```diff
@@ -88,14 +76,58 @@ export default function HeaderCell<R, SR>({
     window.addEventListener('pointermove', onPointerMove);
     window.addEventListener('pointerup', onPointerUp);
   }
+  const sortIndex = sortColumns?.findIndex((sort) => sort.columnKey === column.key);
+  const sortColumn =
+    sortIndex !== undefined && sortIndex > -1 ? sortColumns![sortIndex] : undefined;
+  const sortDirection = sortColumn?.direction;
+  const priority = sortColumn !== undefined && sortColumns!.length > 1 ? sortIndex! + 1 : undefined;
+  const ariaSort =
+    sortDirection && !priority ? (sortDirection === 'ASC' ? 'ascending' : 'descending') : undefined;
+
+  const onSort = (ctrlClick: boolean) => {
+    if (onSortColumnsChange == null) return;
+    const { sortDescendingFirst } = column;
+    if (sortColumn === undefined) {
+      // not currently sorted
+      const nextSort: SortColumn = {
+        columnKey: column.key,
+        direction: sortDescendingFirst ? 'DESC' : 'ASC'
+      };
+      onSortColumnsChange(sortColumns && ctrlClick ? [...sortColumns, nextSort] : [nextSort]);
+    } else {
+      let nextSortColumn: SortColumn | undefined;
+      if (
+        (sortDescendingFirst && sortDirection === 'DESC') ||
+        (!sortDescendingFirst && sortDirection === 'ASC')
+      ) {
+        nextSortColumn = {
+          columnKey: column.key,
+          direction: sortDirection === 'ASC' ? 'DESC' : 'ASC'
+        };
+      }
+      if (ctrlClick) {
+        const nextSortColumns = [...sortColumns!];
+        if (nextSortColumn) {
+          // swap direction
+          nextSortColumns[sortIndex!] = nextSortColumn;
+        } else {
+          // remove sort
+          nextSortColumns.splice(sortIndex!, 1);
```

Comment:
There are many places where we are using `!` and it can be confusing when it not undefined. I usually prefer explicit checks but this is fine for now